libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -86,6 +86,139 @@ method init(g: GossipSub) =
   g.handler = handler
   g.codec = GossipSubCodec
 
+proc replenishFanout(g: GossipSub, topic: string) {.async.} =
```

Comment:
Moved this block of above `rpcHandler` since it now accesses this methods. Forward declarations would give me errors about `gcsafe`ty, didn't wanna deal with it here.